buildmaster/NMCPD/filter.py

```diff
@@ -0,0 +1,64 @@
+#Filter for NMCPD
+
+import yaml
+
+def filter_NMCPD():
+    with open('metadata.yaml', 'r') as file:
+        metadata = yaml.safe_load(file)
+
+    version = metadata['hepdata']['version']
+    tables  = metadata['hepdata']['tables']
+
+    data_central = []
+    kin_avg = []
+    kin_min = []
+    kin_max = []
+    stat_error = []
+    sys_error = []
+    index = 1
+    
+    for i in tables:
+        hepdata_tables="rawdata/HEPData-ins426595-"+version+"-Table_"+str(i)+".yaml"
+        with open(hepdata_tables, 'r') as file:
+            input = yaml.safe_load(file)
+            
+        values = input['dependent_variables'][0]['values']
+        x = float(input['dependent_variables'][0]['qualifiers'][4]['value'])
+        sqrts = float(input['dependent_variables'][0]['qualifiers'][3]['value'])
+        
+        for j in range(len(values)):
+            data_central_value = {'index' : index, 'value' : input['dependent_variables'][0]['values'][j]['value']}      
+            data_central.append(data_central_value)
+            Q2 = input['independent_variables'][0]['values'][j]['value']
+            y = Q2 / ( sqrts * sqrts * x )
+            kin_avg_value = {'index' : index , 'x' : x , 'q2' : Q2 , 'y' : y}
```

Comment:
Are the kinematic values always going to be named x, q2 and y?
(I guess the question is, are x and y standing for "whatever is in the x and y axis and so I should use these two names)

Reply:
NO. This si true for DIS, but NOT for, say, W boson production. Would you prefer a more neutral naming, such as Kin_1, Kin_2 etc?

Reply:
Where the idea is to have as many Kin_* as we want, depending on the nature of the observable?

Reply:
Nono, this is fine. I can just read the name form the file. But if they were going to be always the same I would just hardcode them in the reader.